### ISSUE TITLE
Recorder is not recording clear audio

### ISSUE THREAD

**Issue by RizwanaDesai** [AudioKit/Cookbook#164]
### macOS Version(s) Used to Build

macOS 13 Ventura

### Xcode Version(s)

Xcode 14

### Description

Recorder is not recording clear audio,

We are using AudioKit in our app and from iOS 18 we are facing difficulty with audio quality, tried out Recorder in this demo app too, and demo has the same issue with audio quality.

### Crash Logs, Screenshots or Other Attachments (if applicable)

Steps to reproduce in Demo App:
- Click on Mini Apps
- Click on Recorder
- Record Audio
- And Play that audio, audio sound would not be clear in that.

MacOS:14.2
Xcode: 15.1
Issue reproducible in iOS18 beta

**Comment by RizwanaDesai** [comment]
Yeah, I was about to try the same, as when i tried directly with audio engine it worked with 48000 sample Rate.
Thanks @NickCulbertson!

**Comment by RizwanaDesai** [comment]
`Adding Settings.sampleRate = 48_000` resolves the issue; however, the first recording attempt results in a blank audio file. After navigating back to Mini Apps and attempting to record again, it consistently works from the second attempt onward. @NickCulbertson

**Comment by NickCulbertson** [comment]
Thanks for the info. I'll download the beta this week or next to see what is going on. One thing to note is that if I add `Settings.sampleRate = 48_000` in my current version of the macOS/iOS using Xcode 15 it creates pops. So I would check for backwards compatibility. There might be some way to check the sample rate of the input node directly, but I have not had a chance to try that yet.

I was not able to recreate the issue of the first recording not working. That might have to do with where you are calling `Settings.sampleRate = 48_000`.

**Comment by RizwanaDesai** [comment]
nvm fixed it. Will have to configure setting before audio engine initialize

**Comment by NickCulbertson** [comment]
Okay, I see. In CookbookApp you could update the init method to be: 
```
init() {
        let engine = AudioEngine()
        guard let input = engine.input else {
            fatalError()
        }
        
#if os(iOS)
    do {
        Settings.bufferLength = .short
        Settings.sampleRate = input.avAudioNode.inputFormat(forBus: 0).sampleRate

        try AVAudioSession.sharedInstance().setPreferredIOBufferDuration(Settings.bufferLength.duration)
        try AVAudioSession.sharedInstance().setCategory(.playAndRecord,
                                                        options: [.defaultToSpeaker, .mixWithOthers, .allowBluetoothA2DP])
        try AVAudioSession.sharedInstance().setActive(true)
    } catch let err {
        print(err)
    }
#endif
    }
```
It is not an ideal solution, but it allows us to set the input to mic's format and it doesn't interfere with the Recorder setup.

**Comment by NickCulbertson** [comment]
Thanks for the heads up! This is now fixed here: https://github.com/AudioKit/Cookbook/pull/165

**Comment by NickCulbertson** [comment]
Small update from #167

```
if #available(iOS 18.0, *) {
     Settings.sampleRate = 48_000
}
``` 

This seems to work better than setting it to AVAudioSession's sampleRate.
